### DISCUSSION TITLE
How to handle indentation

### DISCUSSION THREAD

**Discussion by oovm** [badicsalex/peginator#15] [Q&A]
I'm trying to parse a markdown list, but I don't know how to deal with indentation.

```markdown
1. level 1
  - level 2
    still level2 text
    + level 3 
2. level 1
  - level 2
```

It seems that I always need to know the situation of the previous lines, and then judge whether the previous line is an independent list item, and then judge whether this line is indent or dedent

**Comment by badicsalex** [comment]
The way python does it (or at least used to) is that they do a kind of preprocessing. They first go through the code and put special "\~INDENT\~" "\~DEDENT\~" tags to the beginning of the line (if applicable, and being very careful about nesting), and then use these tags in the actual parser:

https://docs.python.org/3/reference/lexical_analysis.html#indentation
